src/utils/display-observer.js

```diff
@@ -178,16 +304,52 @@ export class DisplayObserver {
       const observations = this.targetObservations_.get(target);
       const oldDisplay = computeDisplay(observations, !this.isDocDisplay_);
       const newDisplay = computeDisplay(observations, this.isDocDisplay_);
-      notifyIfChanged(callbacks, newDisplay, oldDisplay);
+      notifyIfChanged(callbacks, target, newDisplay, oldDisplay);
+    });
+  }
+
+  /**
+   * @param {boolean} isDisplayed
+   * @param {!Element} container
+   * @private
+   */
+  containerObserved_(isDisplayed, container) {
```

Comment:
Yeah, not sure about this, tbh. Recreating InOb would definitely have its cost. The issue I'm having: the intersection observer on the container may very well see the intersections - it'd do so unless the container or the observed child are truly set `display:none`. However, from the point of view of the document, the container itself as a whole might be not-displayed. So, I'd then need to override the container observations with `false` somehow. So, what I do here instead is disconnect the observer and instead overwrite all values to `false`, knowing that the observer cannot flip them back to `true` until it's recreated. WDYT?

Reply:
Does the container's InOb not alert all the targets that they are not visible anymore? I had assumed we could make this function just a noop, but if more is required then it makes sense to disconnect as display changes.

Reply:
Not always. It's possible to have a situation where the container's `InOb` considers a target displayable, but the container itself is not. This is because `InOb` doesn't have a concept of "displayable" - it can only answer the question of whether a target "intersects" the container.